File: PiIrrigateServer/PiIrrigateServer/Models/Zone.cs
namespace PiIrrigateServer.Models
{
    public class Zone
    {
        public Guid ZoneId { get; set; } // Unique identifier for the zone
        public string? Name { get; set; } // Name of the zone
        public string? ConnectionString { get; set; } // ConnectionString fro IotHub Device
        public ICollection<Device> Devices { get; set; } = new List<Device>(); // Array of devices in the zone
        public Guid? UserId { get; set; }
        public User User { get; set; } = null!; // Navigation property to the User entity
    }
}

File: PiIrrigateWeb/ClientApp/src/app/models/zone.ts
export interface Zone {
  zoneId: string;                // Unique identifier for the zone (GUID as string)
  name?: string;                 // Name of the zone (optional)
  userId?: string;               // User ID (optional, GUID as string)
}

export interface SensorReading {
  timestamp: Date;         // Timestamp of the reading
  zoneId: string;          // Unique identifier for the zone (GUID as string)
  mac: string;             // Unique identifier for the device
  temperature: number;     // Temperature
  humidity: number;        // Humidity
  soilMoisture: number;    // Soil moisture
  rainfall: number;        // Rainfall
}

export interface Device {
  mac: string;             // Unique identifier for the device (MAC address)
  zoneId: string;          // Unique identifier for the zone (GUID as string)
  name?: string;           // Name of the device (optional)
  location?: string;     // Location of the device (optional)
}